packages/realm_dart/lib/src/handles/native/config_handle.dart

```diff
@@ -86,7 +86,7 @@ class ConfigHandle extends HandleBase<realm_config> {
       } else if (config is InMemoryConfiguration) {
         realmLib.realm_config_set_in_memory(configHandle.pointer, true);
       } else if (config is FlexibleSyncConfiguration) {
-        realmLib.realm_config_set_schema_mode(configHandle.pointer, realm_schema_mode.RLM_SCHEMA_MODE_ADDITIVE_DISCOVERED);
+        realmLib.realm_config_set_schema_mode(configHandle.pointer, realm_schema_mode.RLM_SCHEMA_MODE_ADDITIVE_EXPLICIT);
```

Comment:
This is the actual change, the rest is just cleanup of test code that I noticed while running the tests locally.